### PR TITLE
add job status checks

### PR DESCRIPTION
**Pull request by billiford**
- add job status checks

### CODE REVIEW

File: pkg/kubernetes/job_test.go [hunk 1]
```diff
@@ -84,5 +85,55 @@ var _ = Describe("Job", func() {
 	})
 
 	Describe("#Status", func() {
+		var s manifest.Status
+
+		BeforeEach(func() {
+			o := job.Object()
+			completions := int32(1)
+			o.Status.Succeeded = 1
+			o.Spec.Completions = &completions
```
Comment (yasmin-abdullah) [inline pkg/kubernetes/job_test.go:94]: move completions := int32(1) to the top so you create the object and fill in the values after

File: pkg/kubernetes/job_test.go [hunk 2]
```diff
@@ -84,5 +85,55 @@ var _ = Describe("Job", func() {
...
+
+			When("the job is not finished", func() {
+				It("returns the expected status", func() {
+					Expect(s.Stable.State).To(BeFalse())
+					Expect(s.Stable.Message).To(Equal("Waiting for jobs to finish"))
+				})
+			})
+		})
+
+		When("it succeeeds", func() {
```
Comment (yasmin-abdullah) [inline pkg/kubernetes/job_test.go:132]: I found this confusing because it's at the same level as the context block and the description is ambiguous.

**Review by yasmin-abdullah — CHANGES_REQUESTED** [review]

File: pkg/kubernetes/job.go [hunk 3]
```diff
@@ -9,12 +9,6 @@ import (
 
 type Job interface {
 	Status() manifest.Status
-	// GetObjectMeta() metav1.ObjectMeta
-	// GetPodStatus() v1.PodStatus
-	// GetLabels() map[string]string
-	// GetNamespace() string
-	// GetName() string
-	// GetUID() string
```
[the remark this answers is not in the document]
Comment (billiford) [reply pkg/kubernetes/job.go:17]: We don't need these for a job type *yet*.

File: pkg/kubernetes/job.go [hunk 4]
```diff
@@ -73,5 +67,27 @@ func (j *job) State() string {
 
 func (j *job) Status() manifest.Status {
 	s := manifest.DefaultStatus
+
+	completions := int32(1)
+	spec := j.j.Spec
+	status := j.j.Status
+
+	if spec.Completions != nil {
```
[the remark this answers is not in the document]
Comment (billiford) [reply pkg/kubernetes/job.go:75]: It's a pointer to an int32.

File: pkg/kubernetes/job_test.go (hunk 2 above)
Comment (billiford) [reply pkg/kubernetes/job_test.go:132]: This is just the success case. I typically refer to the happy path as "when it succeeds".

**Review by yasmin-abdullah — APPROVED** [review]

**Pull request merged**